src/Popover.jsx

```diff
@@ -0,0 +1,200 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import cx from 'classnames';
+import bindAll from './utils/bindAll';
+
+/**
+ * @module Popover
+ */
+class Popover extends React.Component {
+	constructor(props) {
+		super(props);
+
+		bindAll(this,
+			'updateFocusBy',
+			'openMenu',
+			'closeMenu',
+			'onClick',
+			'onKeyDown',
+			'onKeyDownMenuItem',
+			'onClick',
+			'onBlur'
+		);
+
+		this.state = {
+			isActive: false,
+			selectedIndex: 0
+		};
+
+		this._menuItems = new Map();
+	}
+
+	updateFocusBy(delta) {
+		const targetIndex = this.state.selectedIndex + delta;
+		if (targetIndex >= 0 && targetIndex < this.menuItems.length) {
+			this.setState({ selectedIndex: targetIndex });
+		}
+	}
+
+	openMenu() {
+		this.setState({ isActive: true });
+	}
+
+	closeMenu() {
+		this.setState({ isActive: false });
+	}
+
+	onBlur() {
+		// On blur, browsers always focus `<body>` before moving focus
+		// to the next actual focused element.
+		//
+		// This zero-length timeout ensures the browser will return the
+		// actual focused element instead of `<body>`
+		window.setTimeout(() => {
+			const focusedOptionClass = document.activeElement.parentNode.classList;
+
+			// don't close the popover if we're moving focus to an menu item
+			if (focusedOptionClass && focusedOptionClass.contains('popover-menu-option')) {
+				return;
+			}
+
+			this.closeMenu();
+		}, 0);
+	}
+
+	onClick(e) {
+		this.openMenu();
+	}
+
+	onKeyDown(e) {
+		switch(e.key) {
+		case 'Enter':
+			if (!this.state.isActive) {
+				this.openMenu();
+			}
+			break;
+		case 'Escape':
+			this.closeMenu();
+			break;
+		}
+	}
+
+	onKeyDownMenuItem(e) {
+		switch(e.key) {
+		case 'ArrowDown':
+			this.updateFocusBy(1);
+			break;
+		case 'ArrowUp':
+			this.updateFocusBy(-1);
+			break;
+		case 'Enter':
+			if (this._menuItems.get(this.state.selectedIndex)) {
+				this._menuItems.get(this.state.selectedIndex).props.onClick(e);
+			}
+			break;
+		}
+	}
+
+	componentDidUpdate() {
+		const selectedItemEl = this._menuItems.get(this.state.selectedIndex);
+		if (selectedItemEl) {
+			ReactDOM.findDOMNode(selectedItemEl).focus();
+		}
+	}
+
+	renderMenuItems() {
+		this.menuItems = this.props.menuItems.map((menuItem, i) => {
+			return (
+				<li
+					key={i}
+					className='popover-menu-option arg'
+					role='menuitem'
+					onKeyDown={this.onKeyDownMenuItem}
+				>
+					{
+						/*
+						* treat each user-provided menu item element as the
+						* keyboard-navigable, focusable 'menuitem' role
+						*/
+						React.cloneElement(menuItem,
+							{
+								tabIndex: '-1',
+								className: 'popover-menu-option-target',
```

Comment:
Setting the className like this overrides the elements className. You can keep the className passed into the prop with:
```
className: `popover-menu-option-target ${menuItem.props.className}`
```

However, I'm curious if we want to prevent that by design. Should we allow users of the Popover component set the className of elements within menu? I'm leaning towards allowing it since it will be useful with SQ2 classes.